### ISSUE TITLE
Email specification at the level of domain

### ISSUE THREAD

**Issue by jennybc** [r-lib/gargle#118]
Idea from @Hadley based on his Google analytics work:

Matching email and pre-approving token discovery in a `PKG_auth()` function could be based on domain, e.g. `domain = "rstudio.com". This is intermediate between `email = TRUE` and having to put a specific email into a script.

This way different people at the same company could share a script, without incessantly editing the `email`.

**Comment by jonthegeek** [comment]
Yes please! I'd be hacking this in myself, would be great if it was in gargle natively!

**Comment by jennybc** [comment]
How should this be implemented?

Proposal for discussion: If `email` starts with `*@` (or maybe just `@`?), then that authorizes a match to any token, with the correct other info (e.g. scopes, oauth app), where `email` is from the specified domain. So something like:

```
sheets_auth(email = "*@example.com")

# or
options(gargle_oauth_email = "*@example.com")
```

**Comment by MarkEdmondson1234** [comment]
I don't understand how this would work with the Google Analytics API for example, as that is only granted on a per email basis?  Wish it would work on a domain level, would make things much easier ;)  But I guess Google Suite stuff like Google Sheets do allow domain level auth?

**Comment by jennybc** [comment]
No, this is just about granting gargle permission to use a token found in the cache. It's a gargle-internal matter.

**Comment by hadley** [comment]
@MarkEdmondson1234 my assumption is that you'd generally be granted to work stuff from your work email address; so (e.g.) I can use `email = "@rstudio.com", and it'll find the right token for any of my co-workers at RStudio
